java/src/com/ibm/streamsx/topology/internal/context/service/RemoteStreamingAnalyticsServiceStreamsContext.java

```diff
@@ -39,10 +39,13 @@ public RemoteStreamingAnalyticsServiceStreamsContext() {
     @Override
     protected Future<BigInteger> action(com.ibm.streamsx.topology.internal.context.JSONStreamsContext.AppEntity entity) throws Exception {
         remoteContext.submit(entity.submission).get();
-        
+
         JsonObject results = object(entity.submission, RemoteContext.SUBMISSION_RESULTS);
         if (results != null) {
-            String jobId = jstring(results, "jobId");
+            // V2 uses id, V1 uses jobId
+            String jobId = jstring(results, "id");
+            if (jobId == null)
+                jobId = jstring(results, "jobId");
             if (jobId != null)
```

Comment:
Ideally, I think this whole chunk should move down into the version-specific implementation, but that looked like it would require more changes to the context than I really wanted at this time.

As it is, it's looking for one of two members but doesn't know which. It could be restructured of course, or even check with `has` before fetching, but it wouldn't be correct to just drop an `else` in before the second `if` because we want that code to execute if `jobId` got set in the first `if`.